src/orquestra/sdk/_base/cli/_ui/_errors.py

```diff
@@ -15,12 +18,52 @@
 from orquestra.sdk.schema.responses import ResponseStatusCode
 
 
-def _print_traceback(e: Exception):
-    # Newer Python versions like 3.10 allow passing just the exception object to
-    # traceback.format_exception(). Python 3.8 requires an explicit 3-argument form.
+def _compact_tb(tb: TracebackType):
+    return "{}:{}:{}".format(
+        tb.tb_frame.f_code.co_name,
+        Path(tb.tb_frame.f_code.co_filename).name,
+        tb.tb_lineno,
+    )
+
+
+def _compact_exc(e: BaseException, prefix: str = ""):
+    tb = e.__traceback__
+    exc_message = f"{e}"
+    spacing = ": " if len(exc_message) > 0 else ""
+    file_details = f"({_compact_tb(tb)})" if tb is not None else ""
+    return "{}[red][b]{}{}[/b]{} {}[/red]".format(
+        prefix,
+        type(e).__name__,
+        spacing,
+        exc_message,
+        file_details,
+    )
+
 
-    tb_lines = traceback.format_exception(type(e), e, e.__traceback__)
-    click.secho("".join(tb_lines), fg="red", file=sys.stderr)
+def _print_traceback(
+    e: BaseException, level: int = 0, console: Optional[rich.console.Console] = None
```

Comment:
**question**: do I understand correctly that `level` and `console` parameters are meant to be _unset_ by external callers, but set when going down in the recursion?

Reply:
Yes, I thought this was an OK interface because they're `_private` functions.

Reply:
Right. I event wanted to suggest a tiny refactor but none of this matters for private functions.